universe/Universe.cpp

```diff
@@ -856,12 +856,17 @@ void Universe::UpdateMeterEstimatesImpl(const std::vector<int>& objects_vec, boo
 
 void Universe::BackPropagateObjectMeters(const std::vector<int>& object_ids) {
     // copy current meter values to initial values
-    for (auto& obj : m_objects.FindObjects(object_ids))
+    for (auto& obj : m_objects.find(object_ids))
         obj->BackPropagateMeters();
 }
 
 void Universe::BackPropagateObjectMeters()
-{ BackPropagateObjectMeters(m_objects.FindObjectIDs()); }
+{
+    std::vector<int> result;
+    for (const auto& obj : m_objects.all())
+        result.push_back(obj->ID());
+    BackPropagateObjectMeters(result);
```

Comment:
This version of `BackPropagateObjectMeters` should probably just do the `obj->BackPropagateMeters()` call directly on all the objects, rather than looping over objects to get their ids, push them into a vector, pass that vector into the overload, which then calls `find` with the vector of ids to get the objects again, in order to do the actual calls.

Reply:
The vector<int> variant is only used at this location so it can be dropped entirely.

Reply:
Done